src/SourceLocator/Type/PhpInternalSourceLocator.php

```diff
@@ -5,31 +5,30 @@
 namespace Roave\BetterReflection\SourceLocator\Type;
 
 use InvalidArgumentException;
-use ReflectionClass;
+use ReflectionClass as CoreReflectionClass;
+use ReflectionFunction as CoreReflectionFunction;
 use Roave\BetterReflection\Identifier\Identifier;
 use Roave\BetterReflection\SourceLocator\Ast\Locator;
 use Roave\BetterReflection\SourceLocator\Exception\InvalidFileLocation;
 use Roave\BetterReflection\SourceLocator\Located\InternalLocatedSource;
 use Roave\BetterReflection\SourceLocator\Located\LocatedSource;
-use Roave\BetterReflection\SourceLocator\Reflection\SourceStubber;
+use Roave\BetterReflection\SourceLocator\SourceStubber\ReflectionSourceStubber;
+use Roave\BetterReflection\SourceLocator\SourceStubber\SourceStubber;
 use function class_exists;
-use function file_get_contents;
+use function function_exists;
 use function interface_exists;
-use function is_file;
-use function is_readable;
-use function preg_match;
 use function trait_exists;
 
 final class PhpInternalSourceLocator extends AbstractSourceLocator
 {
     /** @var SourceStubber */
     private $stubber;
 
-    public function __construct(Locator $astLocator)
+    public function __construct(Locator $astLocator, ?SourceStubber $stubber = null)
     {
         parent::__construct($astLocator);
 
-        $this->stubber = new SourceStubber();
+        $this->stubber = $stubber ?? new ReflectionSourceStubber();
```

Comment:
shouldn't this include the phpstorm stubber as well?

Reply:
@theofidry No, because it would be BC break.